doc/kubernetes/ray-cluster-external-redis.yaml

```diff
@@ -0,0 +1,136 @@
+# Ray head node service, allowing worker pods to discover the head node.
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: ray
+  name: example-cluster-ray-head
+spec:
+  ports:
+  - name: client
+    protocol: TCP
+    port: 10001
+    targetPort: 10001
+  - name: dashboard
+    protocol: TCP
+    port: 8265
+    targetPort: 8265
+  - name: redis
+    protocol: TCP
+    port: 6379
+    targetPort: 6379
+  selector:
+    component: ray-head
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: ray
+  name: ray-head
+spec:
+  # Do not change this - Ray currently only supports one head node per cluster.
+  replicas: 1
+  selector:
+    matchLabels:
+      component: ray-head
+      type: ray
+  template:
+    metadata:
+      labels:
+        component: ray-head
+        type: ray
+    spec:
+      # If the head node goes down, the entire cluster (including all worker
+      # nodes) will go down as well. If you want Kubernetes to bring up a new
+      # head node in this case, set this to "Always," else set it to "Never."
+      restartPolicy: Always
+
+      # This volume allocates shared memory for Ray to use for its plasma
+      # object store. If you do not provide this, Ray will fall back to
+      # /tmp which cause slowdowns if is not a shared memory volume.
+      volumes:
+      - name: dshm
+        emptyDir:
+          medium: Memory
+      containers:
+        - name: ray-head
+          image: rayproject/ray:latest
+          env:
+            # RAY_REDIS_ADDRESS can force ray to use external redis
+            - name: RAY_REDIS_ADDRESS
+              value: redis:6379
+          imagePullPolicy: IfNotPresent
+          command: [ "/bin/bash", "-c", "--" ]
+          args:
+            - "ray start --head --num-cpus=$MY_CPU_REQUEST --object-manager-port=22345 --node-manager-port=22346 --dashboard-host=0.0.0.0 --block"
```

Comment:
```suggestion
            - "ray start --head --port=6380 --num-cpus=$MY_CPU_REQUEST --dashboard-host=0.0.0.0 --block"
```

Reply:
Or even better:
```suggestion
            - "ray start --head --port=$EXAMPLE_CLUSTER_RAY_HEAD_SERVICE_PORT_GCS_SERVER --num-cpus=$MY_CPU_REQUEST --block"
```

Reply:
I used the updated yaml to deploy the cluster. The Ray cluster is functional. But then I checked the external Redis and it seems no additional keys have been added. I am wondering if that is expected.
![Screenshot from 2022-09-21 23-16-46](https://user-images.githubusercontent.com/20148872/191674033-11d94e2f-4d66-4142-9f0d-3220d071dc4d.png)

Reply:
I see, odd. Let's see if we can get some input from Ray core folks.

For ease of reading the config in this PR, are the suggest changes ok to commit?

Reply:
Yeah, they all seem good! I committed on your behalf, thanks!

Reply:
ty